### PR TITLE
add golangci-lint configuration and CI job

### PR DESCRIPTION
**Pull request by twpayne**
This includes a format check that #22 did not include.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+linters:
```
Comment (stapelberg) [inline .golangci.yml:1]: Why this long explicitly-configured-looking list?

Isn’t there a standard set of linters that make sense for most projects?

Comment (twpayne) [reply .golangci.yml:1]: `golangci-lint` used to have an `enable-all` to enable all linters, and then you could explicitly disable the linters you didn't want. However that got removed a few releases ago as it's fragile when you bump the `golangci-lint`: a new linter gets added and suddenly your build starts failing. So, it `enable-all` got removed and you have to explicitly enable the linters you want.

Each linter encodes a certain set of subjective opinions on how Go should be written. Personally, I like to have each linter either explicitly enabled or explicitly disabled for transparency, which is what's in this PR. However, if you'd prefer for the config to be closer to whatever `golangci-lint` enables by default (which is generally only the least contentious linters), then I will happily update the PR to do that.

Comment (stapelberg) [reply .golangci.yml:1]: Thanks for the context.

For this particular repository, I think it’d be better to use the default set of non-contentious linters.

Given the rate of changes, I don’t think manually maintaining linters is a good use of anyone’s time here :)

Comment (twpayne) [reply .golangci.yml:1]: Done. I explicitly enabled `gofmt` (as it's disabled by default and we want to preserve the check from Travis) and disabled `errcheck` as there are five instances of unchecked errors. Everything else is left at `golangci-lint`'s defaults.

**Review by stapelberg — APPROVED** [review]

**Pull request merged**